lib/logstash/inputs/tcp.rb

```diff
@@ -86,48 +83,121 @@ def register
         @ssl_context.cert_store = @cert_store
         @ssl_context.verify_mode = OpenSSL::SSL::VERIFY_PEER|OpenSSL::SSL::VERIFY_FAIL_IF_NO_PEER_CERT
       end
-    end # @ssl_enable
+    end
+
+    # note that since we are opening a socket in register, we must also make sure we close it
```

Comment:
opening it in register makes testing easier, we can create a client socket before running the plugin.